### ISSUE TITLE
[Android] Cores directory path doesnt save properly

### ISSUE THREAD

**Issue by klepp0906** [libretro/RetroArch#9117]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

under settings > directory - changing the path to cores does not retain.  all others seem to, however this one resets after closing and re - opening RA. 

### Expected behavior

the user to be able to define/designate his core path same as the other paths. 

### Actual behavior

It displays as changed, however upon closing and re opening the program, it resets to the default. 

### Steps to reproduce the bug

1. navigate to the directory settings
2. change the core path to anywhere non-default.  i chose emulated/0/retroarch/cores 
3. close and re-open RA.  note the path has reverted. 

### Bisect Results

no idea, just now trying to set up RA on my shields so I can play if my PC is down or my kids can play while im using the pc.  (i usually stream it)

### Version/Commit
You can find this information under Information/System Information

- RetroArch: May 8 2019 - 8c17f12

### Environment information

- OS: Android (Nvidia Shield TV)

**Comment by hizzlekizzle** [comment]
Yes, on Android, programs cannot load dynamic libraries from any directory except for its own root-perms-only directory (in our case, /data/data/com.retroarch/*)

While it doesn't fix your problem, exactly, we should hide that setting on Android, since it can't be changed.

**Comment by klepp0906** [comment]
thank you for letting me know.  Assumed this may be the case.

fortunately the rest of them seem to be okay.

really wish I had rooted the device prior to setting it all up.  that ships sailed now :)  at least then i could access and modify the core files if need be. 

While I have you here,  A minor nitpick.  When changing the temp directory - retroarch will still create a temp directory in the root of emulated/0 upon closing/re-opening RA.

Thanks again for the info, and your time :)

**Comment by hizzlekizzle** [comment]
dunno about the temp thing, but you can delete installed cores by loading one, then go to main menu > information > core information and then choose 'delete core', and you can install new cores by building them or downloading them from the buildbot and then putting them in storage/RetroArch/downloads, then go to 'load core' > install a core and it will copy any *.so libs there into the root-perm core directory without needing root perms.

**Comment by klepp0906** [comment]
yea i just sucked it up and moved the temp directory back to where it was auto creating it anyhow.

no use having 2 temp directories.  I try to keep things organized :P

thanks for the /downloads tip.  that I did not know :)

**Comment by thekiefs** [comment]
See here for a for fixing the directory structure:

https://github.com/libretro/RetroArch/issues/6749

**Comment by LazyBumHorse** [comment]
Oops, wrong number :smile:
I can't take that mention back apparently

**Comment by davidhedlund** [comment]
> See #6749 for fixing the directory structure on Android

This issue (#9117) should be closed as far as I understand, since the core directory is included in #6749. @LibretroAdmin @hizzlekizzle

**Comment by davidhedlund** [comment]
@sonninnos Thanks
